src/fides/api/schemas/connection_configuration/connection_secrets_snowflake.py

```diff
@@ -19,9 +19,19 @@ class SnowflakeSchema(ConnectionConfigSecretsSchema):
         title="Username",
         description="The user account used to authenticate and access the database.",
     )
-    password: str = Field(
+    password: Optional[str] = Field(
         title="Password",
-        description="The password used to authenticate and access the database.",
+        description="The password used to authenticate and access the database. You can use a password or a private key, but not both.",
+        sensitive=True,
+    )
+    private_key: Optional[str] = Field(
+        title="Private key",
+        description="The private key used to authenticate and access the database. It could be encrypted or unencrypted, if encrypted, provide the passphrase.",
+        sensitive=True,
+    )
+    private_key_passphrase: Optional[str] = Field(
+        title="Passphrase",
+        description="The passphrase used for the encrypted private key.",
```

Comment:
Drive by comment - for Pydantic V2, "optional" fields by itself will still be required, it just means that the default is allowed to be none. To keep this optional, can you add a `default=None` field here for both of these new fields.